cpp/ql/src/Likely Bugs/Underspecified Functions/ImplicitFunctionDeclaration.c

```diff
@@ -0,0 +1,8 @@
+/* '#include <stdlib.h>' was forgotton */
+
+int main(void) {
+	/* 'int malloc()' assumed */
+	unsigned char *p = malloc(100);
```

Comment:
Annoyingly, I can't find a compiler that'll actually miscompile this. The three major compilers always keep the return value in a 64-bit register. The Intel compiler will take the return value from the 32-bit `eax` register and sign-extend it but only if the function is called something else than `malloc`!

See https://godbolt.org/z/p9Z5HF.